### PR TITLE
FSET-1696 Progress candidates to sift phase.

### PR DESCRIPTION
**Pull request by benjstephenson**

### CODE REVIEW

File: app/model/ProgressStatuses.scala [hunk 1]
```diff
@@ -121,6 +123,7 @@ object ProgressStatuses {
   case object ASSESSMENT_CENTRE_PASSED_NOTIFIED extends ProgressStatus(ApplicationStatus.ASSESSMENT_CENTRE_PASSED_NOTIFIED)
   case object ASSESSMENT_CENTRE_FAILED_NOTIFIED extends ProgressStatus(ApplicationStatus.ASSESSMENT_CENTRE_FAILED_NOTIFIED)
 
+
```
Comment (henricook) [inline app/model/ProgressStatuses.scala]: You're fairly whitespace happy in this PR, i'm not holding it against you so far

File: app/repositories/sift/ApplicationSiftRepository.scala [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        val phase3PassMarks = testGroupsRoot.getAs[BSONDocument](prevTestGroup).get
+        val phase3Evaluation = phase3PassMarks.getAs[PassmarkEvaluation]("evaluation").get
+        ApplicationForSift(applicationId, phase3Evaluation)
+      }
+    }
+  }
+
+  def progressApplicationToSiftStage(application: ApplicationForSift): Future[Unit] = {
+    val query = BSONDocument("applicationId" -> application.applicationId) ++ eligibleForSiftQuery
+    val update = BSONDocument("applicationStatus" -> ApplicationStatus.SIFT)
```
Comment (henricook) [inline app/repositories/sift/ApplicationSiftRepository.scala]: I'd set a matching progress status (and crucially the date) here so we know WHEN they moved to sift

File: conf/application.conf [hunk 3]
```diff
@@ -359,6 +359,18 @@ scheduling {
       yamlFilePath = locations-and-venues.yaml
     }
   }
+  progress-to-sift-job {
+    enabled = false
+    lockId = "progress-to-sift-job-lock-coordinator"
+    initialDelaySecs = 30
+    intervalSecs = 30
+  }
+  progress-to-assessment-centre-job {
```
Comment (henricook) [inline conf/application.conf]: This looks out of place?

File: it/repositories/CommonRepository.scala [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package repositories
 
-import config.{ CubiksGatewayConfig, LaunchpadGatewayConfig }
+import config.{CubiksGatewayConfig, LaunchpadGatewayConfig}
```
Comment (henricook) [inline it/repositories/CommonRepository.scala]: Someone's got bad brackets config

File: it/repositories/CommonRepository.scala [hunk 5]
```diff
@@ -100,6 +105,31 @@ trait CommonRepository {
...
 
+  def insertApplicationWithPhase3TestNotifiedResults(appId: String, scheme: SchemeId, result: EvaluationResults.Result,
+                             videoInterviewScore: Option[Double] = None,
+                             applicationRoute: ApplicationRoute = ApplicationRoute.Faststream
+                            ): Future[Unit] = {
+    val phase3PassMarkEvaluation = PassmarkEvaluation("", Some(""),
+      List(SchemeEvaluationResult(scheme,
+        result.toPassmark)), "", Some(""))
+
+    assertNotNull("Phase3 pass mark evaluation must be set", phase3PassMarkEvaluation)
```
Comment (henricook) [inline it/repositories/CommonRepository.scala]: Can this ever be null?

File: conf/application.conf (hunk 3 above)
Comment (benjstephenson) [inline conf/application.conf]: This should have been in a separate PR

File: app/repositories/sift/ApplicationSiftRepository.scala (hunk 2 above)
Comment (benjstephenson) [inline app/repositories/sift/ApplicationSiftRepository.scala]: Agree on storing the transition date

File: it/repositories/CommonRepository.scala (hunk 4 above)
Comment (benjstephenson) [inline it/repositories/CommonRepository.scala]: Looking at you @andrewrembrandt

File: it/repositories/sift/ApplicationSiftRepositorySpec.scala [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+      val appsForSift = repository.nextApplicationsForSiftStage(10).futureValue
+      appsForSift mustBe List(
+        ApplicationForSift("appId1", PassmarkEvaluation("", Some(""),
+          List(SchemeEvaluationResult(SchemeId("Commercial"), EvaluationResults.Green.toPassmark)), "", Some(""))),
+        ApplicationForSift("appId4", PassmarkEvaluation("", Some(""),
+          List(SchemeEvaluationResult(SchemeId("Project Delivery"), EvaluationResults.Green.toPassmark)), "", Some(""))))
+    }
+
+    """return no results when there are only phase 3 applications that aren't in Passed_Notified which apply for sift or don't have Green/Passed
```
Comment (henricook) [inline it/repositories/sift/ApplicationSiftRepositorySpec.scala]: I think if you let intellij do its standard new line stuff it'll close your quotes on the line above and open them again on the line below - having to add stripMargin smells a bit

File: it/repositories/CommonRepository.scala (hunk 5 above)
Comment (andrewrembrandt) [inline it/repositories/CommonRepository.scala]: Removed all null checks

File: it/repositories/sift/ApplicationSiftRepositorySpec.scala (hunk 6 above)
Comment (andrewrembrandt) [inline it/repositories/sift/ApplicationSiftRepositorySpec.scala]: That's the intellij form - I've replaced with string concatenation and parens

File: conf/application.conf (hunk 3 above)
Comment (andrewrembrandt) [inline conf/application.conf]: Removed

File: app/model/SerialUpdateResult.scala [hunk 7]
```diff
@@ -37,24 +36,18 @@ case class SerialUpdateResult[UpdateRequest](
...
+    val SUCCESS, FAILURE, PARTIAL, NO_OP = Value
   }
 
   def fromEither[T](results: Seq[Either[T, T]]): SerialUpdateResult[T] = {
 
-    results.foldLeft(List.empty[T], List.empty[T]){ case (acc, res) =>
+    val (f, s) = results.foldLeft(List.empty[T], List.empty[T]){ case (acc, res) =>
       res match {
         case Left(l) => (acc._1 :+ l, acc._2)
         case Right(r) => (acc._1, acc._2 :+ r)
```
Comment (kjharkin) [inline app/model/SerialUpdateResult.scala]: Would deconstructing the ```acc``` into the two constituent parts make this rather obscure class more readable?

File: app/controllers/SiftingController.scala [hunk 8]
```diff
@@ -16,38 +16,35 @@
...
-  def findApplicationsReadyForSifting(schemeId: String): Action[AnyContent] = Action.async { implicit request =>
-    siftAppRepository.findApplicationsReadyForSifting(SchemeId(schemeId)).map { candidates =>
+  def findApplicationsReadyForSchemeSifting(schemeId: String): Action[AnyContent] = Action.async { implicit request =>
+    siftService.findApplicationsReadyForSchemeSift(SchemeId(schemeId)).map { candidates =>
       Ok(Json.toJson(candidates))
     }
   }
 
   def siftCandidateApplication: Action[JsValue] = Action.async(parse.json) { implicit request =>
     withJsonBody[ApplicationSifting] { sift =>
```
Comment (kjharkin) [inline app/controllers/SiftingController.scala]: ApplicationSifting should be renamed to ApplicationForSifting

**Pull request merged**